### DISCUSSION TITLE
Update `renovate/artifacts` with success status so failed artifact updates can block automerge

### DISCUSSION THREAD

**Discussion by narwold** [renovatebot/renovate#16053] [Suggest an Idea]
I'm having an issue where if a PR fails to update its artifacts (especially due to peer dependency issues), it will report such on the PR, and Github will make it look like CI has failed... but if I don't have `renovate/artifacts` as a required CI step in the repo's settings, it won't block the PR from automerging (using platform automerge). If I DO require `renovate/artifacts` as a required steps, it never reports a success status and therefore is pending on any PRs where it doesn't fail.

Would it be possible to make it report a success status when the artifact updating succeeds, or is there another way this should be approached?

**Comment by rarkins** [comment]
So you'd like `renovate/artifacts` set to green for every PR? And it would need to include PRs with no artifacts updating at all too, right?

**Reply by narwold** [reply]
Yes. It's not ideal, but it's better than letting the PR pass CI and automerge when `renovate/artifacts` fails because we don't include it as a required check. Not sure if you'd suggest some other method.

**Reply by viceice** [reply]
Your CI should fail required build step on artifact errors, that would also block automerge 🙃

**Reply by narwold** [reply]
True; that is generally the case. I guess this came up because I was testing on a brand new repo with minimal CI, and I would have thought that there would have been a built-in way to prevent automerge when artifact update fails. If the answer is just "always run another install in CI", that's what we'll have to live with; it just seemed a little off.

**Reply by narwold** [reply]
By the way, thanks for all the work y'all put in on this project. Much appreciated.

**Reply by rarkins** [reply]
So I guess the feature request will be for an option to always set `renovate/artifacts` no matter what.

I'm wondering if instead we should introduce a new, optional generic status check like `renovate/successful-update` which is set to true normally but in cases where artifacts fail you get `renovate/successful-update` failing plus `renovate/artifacts` failing.

**Reply by narwold** [reply]
That makes sense. Thanks so much for helping to refine this.

**Reply by pan-daniel** [reply]
Hello, just wanted to chime in and ask if that feature is anywhere down the pipeline? 

We have ran into exactly the same issue. We do trunk based development outside of GitHub, so we do not have any sort of testing in it. Automerging is safe for us, since the pipeline will fail anyway requiring us to act, but unfortunately we have noticed that Renovatebot has a tendency to merge the PRs when the `renovate/artifacts` check fails and that causes build problems in a few projects. 

In hindsight it's obvious - we should make the PR not mergable in the GitHub itself if this status fails, but as mentioned here the same `renovate/artifacts` check never returns when everything is fine, so the PR just gets stuck.

**Reply by rarkins** [reply]
Important points to note:
1. Even if Renovate always sets this status check, you can't set it to mandatory for all branches because not all branches are created by Renovate (assuming you have user commits too)
2. Renovate won't automerge if artifacts failed, but GitHub's platform automerge might depending on how you have configured automerge rules. You might be better setting platformAutomerge=false and waiting for Renovate to automerge passing branches itself instead of if GitHub merges them

I've created feature request #24138 - please continue any discussions there

**Reply by pan-daniel** [reply]
Thank you very much for your response and for creating the feature request, especially during your holidays.

I will try to set the `platformAutomerge` to false. Maybe that indeed will be a better solution for us.
